includes/class-wc-post-data.php

```diff
@@ -28,10 +28,8 @@ class WC_Post_Data {
 	public static function init() {
 		add_filter( 'post_type_link', array( __CLASS__, 'variation_post_link' ), 10, 2 );
 		add_action( 'shutdown', array( __CLASS__, 'do_deferred_product_sync' ), 10 );
-		add_action( 'set_object_terms', array( __CLASS__, 'set_object_terms' ), 10, 6 );
 		add_action( 'set_object_terms', array( __CLASS__, 'force_default_term' ), 10, 5 );
-
-		add_action( 'transition_post_status', array( __CLASS__, 'transition_post_status' ), 10, 3 );
+		add_action( 'set_object_terms', array( __CLASS__, 'delete_product_query_transients' ) );
```

Comment:
This can get called for different objects, is there a good reason to get rid of filtering in `set_object_terms`?

E.g. with wc-admin active `delete_product_query_transients` gets called for all scheduled actions, which can be e.g. on each order update:
```
 set_object_terms; object id 23513
 set_object_terms; object id 23515
 set_object_terms; object id 11048
 set_object_terms; object is a product
 set_object_terms; object id 11048
 set_object_terms; object is a product
 set_object_terms; object id 23516 \
 set_object_terms; object id 23517  |
 set_object_terms; object id 23518  |
 set_object_terms; object id 23519  |
 set_object_terms; object id 23520  | these are all scheduled actions, where transients would get read & set now
 set_object_terms; object id 23521  |
 set_object_terms; object id 23522  |
 set_object_terms; object id 23523  |
 set_object_terms; object id 23524 /
...
```

Reply:
@peterfabian It removes the need to get the post_type (`get_post_type( $object_id )`) each time, and if you look at `delete_product_query_transients`, all it does is increment a cache prefix:

```
public static function delete_product_query_transients() {
		WC_Cache_Helper::get_transient_version( 'product_query', true );
}
```

This in itself is not query intensive. Seemed better to just allow that to increment rather than have `get_post` running every time terms are set.

Reply:
You're right, did a quick timing exercise, seems like `get_transient_version` is ~6 times faster than `get_post_type`.